velox/dwio/dwrf/common/CacheInputStream.cpp

```diff
@@ -157,6 +158,65 @@ void CacheInputStream::loadSync(dwio::common::Region region) {
     if (entry->isExclusive()) {
       entry->setGroupId(groupId_);
       entry->setTrackingId(trackingId_);
+      auto ssdCache = cache_->ssdCache();
+
+      if (ssdCache) {
+        auto& file = ssdCache->file(fileNum_);
+        auto ssdPin =
+            file.find(cache::RawFileCacheKey{fileNum_, region.offset});
+        if (!ssdPin.empty() && ssdPin.run().size() < entry->size()) {
+          LOG(INFO) << fmt::format(
+              "IOERR: Ssd entry for {}:{} {} b shorter than requested {}b",
+              entry->key().fileNum.id(),
+              entry->key().offset,
+              ssdPin.run().size(),
+              entry->size());
+          ssdPin.clear();
+        }
+        if (!ssdPin.empty()) {
+          uint64_t usec = 0;
+          // SsdFile::load wants vectors of pins. Put the pins in a
+          // temp vector and then put 'pin_' back in 'this'. 'pin_'
+          // is exclusive and not movable.
+          std::vector<cache::SsdPin> ssdPins;
+          ssdPins.push_back(std::move(ssdPin));
+          std::vector<cache::CachePin> pins;
+          pins.push_back(std::move(pin_));
+          bool loadedFromSsd = false;
+          try {
+            MicrosecondTimer timer(&usec);
+            file.load(ssdPins, pins);
+            loadedFromSsd = true;
+          } catch (const std::exception& e) {
+            try {
+              LOG(ERROR)
+                  << "IOERR: Failed SSD loadSync. offset = "
+                  << entry->key().offset << " " << e.what() << " "
+                  << process::TraceContext::statusLine()
+                  << fmt::format(
+                         "stream region {} {}b, start of load {} file {} {}",
+                         region_.offset,
+                         region_.length,
+                         region.offset - region_.offset,
+                         fileNum_,
+                         fileIds().string(fileNum_));
+              // Remove the non-loadable entry so that next access goes to
```

Comment:
Why is it Ok to ignore this error? It would be nice to describe the error handing strategy in the PR description.